internal/v1/handler_blueprints.go

```diff
@@ -149,22 +155,45 @@ func BlueprintFromAPI(cbr CreateBlueprintRequest) (BlueprintBody, error) {
 }
 
 // Util function used to create Blueprint sctruct from DB entry (READ)
-func BlueprintFromEntry(be *db.BlueprintEntry) (BlueprintBody, error) {
+func BlueprintFromEntry(be *db.BlueprintEntry, options ...BlueprintBodyOption) (BlueprintBody, error) {
 	var result BlueprintBody
 	err := json.Unmarshal(be.Body, &result)
 	if err != nil {
 		return BlueprintBody{}, err
 	}
+
+	for _, option := range options {
+		option(&result)
+	}
+
 	return result, nil
 }
 
-func BlueprintFromEntryWithRedactedPasswords(be *db.BlueprintEntry) (BlueprintBody, error) {
-	result, err := BlueprintFromEntry(be)
-	if err != nil {
-		return BlueprintBody{}, err
+func WithRedactedPasswords() BlueprintBodyOption {
+	return func(bp *BlueprintBody) {
+		bp.RedactPasswords()
+	}
+}
+
+func WithRedactedCertificates() BlueprintBodyOption {
+	return func(bp *BlueprintBody) {
+		bp.RedactCertificates()
+	}
+}
+
+func WithRedactedFiles(paths []string) BlueprintBodyOption {
+	return func(bp *BlueprintBody) {
+		if bp.Customizations.Files != nil {
+			files := slices.DeleteFunc(*bp.Customizations.Files, func(file File) bool {
+				return slices.Contains(paths, file.Path)
+			})
+			if len(files) == 0 {
+				bp.Customizations.Files = nil
+			} else {
+				bp.Customizations.Files = &files
+			}
+		}
```

Comment:
I would probably drop the intermediate value `files` deleting directly so the else branch could be dropped, but this is fine.

Reply:
Fwiw, I agree (but also think its not very important)